openpype/hosts/houdini/plugins/publish/validate_review_colorspace.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+import pyblish.api
+from openpype.pipeline import PublishValidationError
+from openpype.pipeline.publish import RepairAction
+from openpype.hosts.houdini.api.action import SelectROPAction
+
+
+class SetDefaultViewSpaceAction(RepairAction):
+    label = "Set default view space"
+    icon = "mdi.monitor"
+
+
+class ValidateReviewColorspace(pyblish.api.InstancePlugin):
+    """Validate Review Colorspace parameters.
+
+    It checks if 'OCIO Colorspace' parameter was set to valid value.
+    """
+
+    order = pyblish.api.ValidatorOrder + 0.1
+    families = ["review"]
+    hosts = ["houdini"]
+    label = "Validate Review Colorspace"
+    actions = [SetDefaultViewSpaceAction, SelectROPAction]
+
+    def process(self, instance):
+        invalid = self.get_invalid(instance)
+        if invalid:
+            raise PublishValidationError(
+                ("'OCIO Colorspace' parameter is not valid."),
+                title=self.label
+            )
+
+    @classmethod
+    def get_invalid(cls, instance):
+        import hou  # noqa
+        import os
+
+        rop_node = hou.node(instance.data["instance_node"])
+        if os.getenv("OCIO") is None:
+            cls.log.warning(
+                "Default Houdini colorspace is used, "
+                " skipping check.."
+            )
+            return
+
+        if rop_node.evalParm("colorcorrect") != 2:
+            # any colorspace settings other than default requires
+            # 'Color Correct' parm to be set to 'OpenColorIO'
+            rop_node.setParms({"colorcorrect": 2})
+            cls.log.debug(
+                "'Color Correct' parm on '%s' has been set to"
+                " 'OpenColorIO'", rop_node
+            )
+
+        if rop_node.evalParm("ociocolorspace") not in \
+                hou.Color.ocio_spaces():
+
+            cls.log.error(
+                "'OCIO Colorspace' value on '%s' is not valid, "
+                "select a valid option from the dropdown menu.",
+                rop_node
+            )
+            return rop_node
+
+    @classmethod
+    def repair(cls, instance):
+        """Set Default View Space Action.
+
+        It is a helper action more than a repair action,
+        used to set colorspace on opengl node to the default view.
+        """
+
+        import hou
+        from openpype.pipeline.colorspace import get_display_view_colorspace_name #noqa
+        from openpype.hosts.houdini.api.lib import get_color_management_preferences #noqa
+
+        rop_node = hou.node(instance.data["instance_node"])
+
+        data =  get_color_management_preferences()
+        config_path = data.get("config")
+        display =  data.get("display")
+        view =  data.get("view")
+
+        cls.log.debug("Get default view colorspace name..")
+
+        default_view_space = get_display_view_colorspace_name(config_path,
```

Comment:
I apologize for my lack of Houdini expertise, but I have a question that may seem basic. 

Is it always necessary for the ROP node to have its colorspace set to display and view? Is there no way to assign display and view names directly to the node instead of setting its parent colorspace? I ask because certain views may be associated with a specific look rather than a particular output colorspace.

Reply:
In houdini review rop I have these options, and I thought that validating `OCIO Colorspace` leads to better color management. 
![image](https://github.com/ynput/OpenPype/assets/20871534/7e1fc339-1e01-48e6-8bbf-dd796ab35e99)
in this zip there's an exr that corresponds to each setting above 
[houdini_review_colorspaces.zip](https://github.com/ynput/OpenPype/files/12148508/houdini_review_colorspaces.zip)

`OCIO Colorspace` only accepts colorspace names 
![image](https://github.com/ynput/OpenPype/assets/20871534/219e6c04-b446-4dc0-99a3-232bd305a093)

the validator itself doesn't force any values, it only checks if  `OCIO Colorspace` was set to a valid value (i.e. "sRGB" is not valid because it is not a colorspace name unlike "Output - sRGB")

and there's an action button that suggestd to use the default color space of the default view and display if user is not sure which one to use. 
![image](https://github.com/ynput/OpenPype/assets/20871534/0240eefd-4b07-47d2-a297-e8dc7d8e8f5e)